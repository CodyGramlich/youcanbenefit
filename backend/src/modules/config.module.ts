import { Module } from '@nestjs/common';
import { ConstantsReadonly } from './constants.readonly'

@Module({
  providers: [
    {
      provide: ConstantsReadonly,
      useValue: new ConstantsReadonly(),
    },
  ],
  exports: [ConstantsReadonly],
})
export class ConfigModule {}