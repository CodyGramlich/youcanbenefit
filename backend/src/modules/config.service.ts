import { Injectable } from '@nestjs/common'

export interface EnvConfig {
  [key: string]: string;
}

@Injectable()
export class ConfigService {
  private readonly envConfig: EnvConfig;

  constructor() {
    this.envConfig = process.env
  }

  get(key: string): string {
    return this.envConfig[key];
  }
}