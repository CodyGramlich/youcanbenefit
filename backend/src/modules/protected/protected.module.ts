import { Module } from '@nestjs/common';
import { ProgramModule } from '../Program'
import { QueryModule } from "../query";
import { ProtectedController } from "./protected.controller";
import { KeyModule } from "../key";
import { ScreenerModule } from "../screener";

@Module({
    imports: [
        ProgramModule,
        QueryModule,
        KeyModule,
        ScreenerModule
    ],
    controllers: [ ProtectedController ],
    providers: [ ProtectedController ]
})
export class ProtectedModule {}
