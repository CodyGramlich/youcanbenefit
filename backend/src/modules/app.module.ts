import { Module, MiddlewareConsumer, RequestMethod } from '@nestjs/common';
import { CorsMiddleware } from '../middleware/cors.middleware'
import { AppController } from './app.controller';
import { ProgramModule } from "./Program"
import { QueryModule } from "./query"
import { KeyModule } from "./key"
import { ProtectedModule } from "./protected"
import { ApiModule } from "./api"
import { DataModule } from './data/data.module';
import { LogModule } from './log/log.module'
import { ConfigModule } from './config.module'

@Module({
  imports: [
    ConfigModule,
    ProgramModule,
    QueryModule,
    KeyModule,
    ProtectedModule,
    ApiModule,
    DataModule,
    LogModule
  ],
  controllers: [AppController]
})
export class ApplicationModule {
  configure(consumer: MiddlewareConsumer) {
    consumer.apply(CorsMiddleware).forRoutes({
      path: '*',
      method: RequestMethod.ALL
    })
  }
}
