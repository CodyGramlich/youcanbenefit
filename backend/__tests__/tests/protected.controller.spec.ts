import * as request from 'supertest';
import { Test } from '@nestjs/testing';
import { ProtectedModule } from '../../src/modules/protected'
import { INestApplication } from '@nestjs/common';

describe('/protected', () => {
  let app: INestApplication;

  beforeAll(async () => {
    const module = await Test.createTestingModule({
      imports: [ProtectedModule]
    })
    .compile();
    app = module.createNestApplication();
    await app.init();
  })

  describe('/login', () => {
    it('GET /login will allow you to login, auth is handled by reverse proxy', () => {
      return request(app.getHttpServer())
        .get('/protected/login/')
        .expect(200)
        .expect(res => {
          expect(res.body.created).toBe(true)
        });
    });
  });

  describe('/program -- CRUD with program datastructures', () => {
    it(' GET /program will return all programs and their queries', () => {
      return request(app.getHttpServer())
        .get('/protected/program/')
        .expect(200)
        .expect(response => {
            const extractedProperties = response.body.map( applicationProgramDto => applicationProgramDto.guid);
            extractedProperties.sort( (a, b) => a.localeCompare(b));

            expect(extractedProperties).toMatchSnapshot();
        })
    });
  });

  afterAll(async () => {
    await app.close();
  });
});