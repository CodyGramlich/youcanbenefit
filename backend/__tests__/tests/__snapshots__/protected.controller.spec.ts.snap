// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/protected /program -- CRUD with program datastructures  GET /program will return all programs and their queries 1`] = `
Array [
  "1iXreywnofiF2jKKknlbKRXAtX",
  "1lKYPN8nNaAMldGufGZNEwfjIK",
  "5b1J8A2fhE9cwVo9W5MfbLmuNr",
  "7Vo9xE20WJJJ1ZORDlog1odyzo",
  "8Lqu2z4WCAP9w5sjhcAeOKuHN5",
  "8U2rl4Ha5GXBVqjNStudvzHIch",
  "9dHBCUXa0z8tEZg6QcYMBJVXjf",
  "9rRXexrA3JzmXbFdKCmjtTj57l",
  "BcKJhMTErAsHfXq1jzealos8kp",
  "bzf4ryrVVsKkqpsEA5dBZd2kP5",
  "EY1QV5T6AyB4Ykl2Lt1Nt9kf9g",
  "GiufEa3ziFwfBEgkjI2gIlfNkk",
  "HiGYI2bhBoQ6n82R9ptoNZ0oaQ",
  "hp5mupqcVY8ZMiKg7Q91Uoi4Wf",
  "iHPbkwKFsck9ZLnMePKSj5e5mz",
  "IO2YsSRkLCIZwIcYGjO6rNkxUE",
  "JIcpL1cap5Eg0ikPk2tecp6JG7",
  "K08iJHYVRGL5pBOomOvwa2zwX2",
  "kc0ov01FtBgr88C5P6EE6lnC8s",
  "NMh2Fu1IG5XnpOf5b6zu2t1Hh6",
  "nQXxT1OMhPGbkqkPHc2QTpofRv",
  "NTVwRTvMb82TIB5VLbf3v36tZN",
  "OGR28icdrofstoYETvIGg19XAv",
  "oZ68OrZFpvkMMGHqV3CFELhuVY",
  "sgKMX0336kgepHbPeObRLOvcFL",
  "tWucjcIELoF0O1Z0DksuXWDGY7",
  "UbdhHxTUe3N5zm5BxR6WQ2i6vf",
  "Y5KEzulkwxviOuzB4Ny5QTzMGD",
  "ZPNzWEr5XfUq5aYdy2bSYIKO5T",
  "zXRcJxGzIrwhmk5LTbE4jdwBMh",
]
`;
